### PR TITLE
Added ContentStatus Deleted

### PR DESCRIPTION
**Pull request by Musaevt**
Only Supervisor can get information of Deleted content;
Edit or RePublish of deleted content Not Allowed !

Fixed Bug: In old version was possible to create a content with Status ALL.
Fixed Bug: In old version was no validation for GetIds requests

### CODE REVIEW

File: HiP-DataStore/Core/ReadModel/CacheDatabaseManager.cs [hunk 1]
```diff
@@ -93,7 +93,10 @@ private void ApplyEvent(IEvent ev)
                     break;
 
                 case ExhibitDeleted e:
-                    _db.GetCollection<Exhibit>(ResourceType.Exhibit.Name).DeleteOne(x => x.Id == e.Id);
+                    var deleteExhibit = _db.GetCollection<Exhibit>(ResourceType.Exhibit.Name).AsQueryable().First(x => x.Id == e.Id);
```
Comment (SvenEV) [inline HiP-DataStore/Core/ReadModel/CacheDatabaseManager.cs]: The code for `*Deleted`-events is quite repetitive, maybe we can extract that into a method like:
```C#
private void MarkDeleted<T>(ResourceType resourceType, int entityId) where T : ContentBase
{
    var collection = _db.GetCollection<T>(resourceType.Name);
    var entity = collection.AsQueryable().First(x => x.Id == entityId);
    entity.Status = ContentStatus.Deleted;
    collection.ReplaceOne(x => x.Id == entityId, entity);           
}
```

File: HiP-DataStore/Core/WriteModel/EntityIndex.cs [hunk 2]
```diff
@@ -78,7 +78,9 @@ public IReadOnlyCollection<int> AllIds(ResourceType entityType, ContentStatus st
                     .AsQueryable()
                     .FilterIf(!isAllowedGetAll, x =>
                         ((status == ContentStatus.All) && (x.Value.Status == ContentStatus.Published)) || (x.Value.UserId == userId))
-                    .Where(x => status == ContentStatus.All || x.Value.Status == status)
+                    .FilterIf(status == ContentStatus.All && !UserPermissions.IsAllowedToGetDeleted(User),
```
Comment (SvenEV) [inline HiP-DataStore/Core/WriteModel/EntityIndex.cs:81]: I think these `FilterIf(...)`-calls could be integrated into the `FilterByStatus(...)` extension method

**Review by SvenEV — CHANGES_REQUESTED** [review]

File: HiP-DataStore/Controllers/ExhibitsController.cs [hunk 3]
```diff
@@ -99,7 +108,7 @@ public IActionResult GetById(int id, DateTimeOffset? timestamp = null)
             if (!ModelState.IsValid)
                 return BadRequest(ModelState);
 
-            var status = _entityIndex.Status(ResourceType.Exhibit, id) ?? ContentStatus.Published;
+            var status = _entityIndex.Status(ResourceType.Exhibit, id) ?? ContentStatus.Deleted;
```
Comment (SvenEV) [inline HiP-DataStore/Controllers/ExhibitsController.cs]: I think defaulting to `ContentStatus.Deleted` is not perfect here: If you try to get an exhibit that doesn't exist, you now always get `403 Forbidden` instead of `404 Not Found`. I find this a bit misleading. Defaulting to `ContentStatus.Published` seems more reasonable, why did you change it?

(this probably applies to pages/routes/etc. as well, I didn't check that)

File: HiP-DataStore/Controllers/ExhibitPagesController.cs [hunk 4]
```diff
@@ -48,12 +48,15 @@ public class ExhibitPagesController : Controller
...
         [ProducesResponseType(typeof(IReadOnlyCollection<int>), 200)]
-        public IActionResult GetAllIds(ContentStatus? status)
+        public IActionResult GetAllIds(ContentStatus status = ContentStatus.Published)
         {
             if (!ModelState.IsValid)
                 return BadRequest(ModelState);
 
-            return Ok(_entityIndex.AllIds(ResourceType.ExhibitPage, status ?? ContentStatus.Published, User.Identity));
+            if (status == ContentStatus.Deleted && !UserPermissions.IsAllowedToGetDeleted(User.Identity))
+                return Forbid();
```
Comment (SvenEV) [inline HiP-DataStore/Controllers/ExhibitPagesController.cs:59]: In many locations (like this one), `[ProducesResponseType(403)]` is missing

**Review by SvenEV — CHANGES_REQUESTED** [review]

File: HiP-DataStore/Controllers/ExhibitPagesController.cs (hunk 4 above)
Comment (Musaevt) [reply HiP-DataStore/Controllers/ExhibitPagesController.cs:59]: You are right, but this annotation is for swagger and swagger can detect forbid(403) automatically

**Review by SvenEV — APPROVED** [review]
